api/v1beta2/foundationdb_database_configuration.go

```diff
@@ -454,6 +510,28 @@ func (configuration DatabaseConfiguration) getRegionPriorities() map[string]int
 	return priorities
 }
 
+// AreSeparatedProxiesConfigured returns true if grv_proxies and
+// commit_proxies are greater than 0 (explicitly set) and Proxies is set
+// to 0
+func (configuration DatabaseConfiguration) AreSeparatedProxiesConfigured() bool {
+	counts := configuration.RoleCounts
+	if counts.Proxies == 0 && (counts.GrvProxies > 0 && counts.CommitProxies > 0) {
+		return true
+	}
+	return false
+}
+
+func (configuration DatabaseConfiguration) GetProxiesString(version Version) string {
+	counts := configuration.GetRoleCountsWithDefaults(DesiredFaultTolerance(configuration.RedundancyMode))
+	var proxies_string string
+	if version.HasSeparatedProxies() && configuration.AreSeparatedProxiesConfigured() {
+		proxies_string += fmt.Sprintf(" commit_proxies=%d grv_proxies=%d", counts.CommitProxies, counts.GrvProxies)
+	} else {
+		proxies_string += fmt.Sprintf(" proxies=%d", counts.Proxies)
+	}
+	return proxies_string
```

Comment:
```golang
 	if version.HasSeparatedProxies() && configuration.AreSeparatedProxiesConfigured() {
 		return fmt.Sprintf(" commit_proxies=%d grv_proxies=%d", counts.CommitProxies, counts.GrvProxies)
 	} 

        return fmt.Sprintf(" proxies=%d", counts.Proxies)
```

We can directly return the generated string instead of appending it to the empty variable.

Reply:
Done.